dynamiqs/sparse.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import annotations
+
+import equinox as eqx
+import jax.numpy as jnp
+from jaxtyping import Array, ArrayLike, ScalarLike
+
+from .qarray import QArray
+
+
+class SparseQArray(QArray):
+    diags: Array
+    offsets: tuple[int, ...] = eqx.field(static=True)
+    dims: tuple[int, ...] = eqx.field(static=True)
```

Comment:
You probably don't want these fields to be static, it'll force jax to recompile for each array you pass

Reply:
Yes let's talk about it

Reply:
Probably we'll start by keeping `offsets` static, and think about making them dynamic later on. It's not so clear yet whether this is doable with `jax.dynamic_slice` or not (because the function still requires a static size as input).